tests/Functional/CorporateFooterRenderTest.php

```diff
@@ -0,0 +1,435 @@
+   */
+  public function testEuFooterCoreBlockRendering(): void {
+    $this->configFactory->getEditable('oe_theme.settings')->set('component_library', 'eu')->save();
+
+    $data = $this->getFixtureContent('eu_footer.yml');
+    $this->renderCorporateBlocksFooter('eu', $data);
+
+    $this->getSession()->reload();
+    $assert = $this->assertSession();
+
+    // Make sure that footer block is present.
+    $this->assertFooterPresence('core', 6);
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section1');
+
+    // Site owner is not set yet, lets make sure we don't have a description.
+    $assert->elementNotExists('css', 'div.ecl-footer-core__description');
+
+    // Assert presence of ecl logo in footer.
+    $this->assertEclLogoPresence($section, 'core');
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section3 .ecl-footer-core__section:nth-child(1)');
+
+    $actual = $section->find('css', '.ecl-footer-core__title');
+    $this->assertEquals('Contact title', $actual->getText());
+
+    $items = $data['contact'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section3 .ecl-footer-core__section:nth-child(2)');
+
+    $actual = $section->find('css', '.ecl-footer-core__title');
+    $this->assertEquals('Social media title', $actual->getText());
+
+    $items = $data['social_media'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section3 .ecl-footer-core__section:nth-child(3)');
+
+    $actual = $section->find('css', '.ecl-footer-core__title');
+    $this->assertEquals('Legal links title', $actual->getText());
+
+    $items = $data['legal_links'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-core section.ecl-footer-core__section4');
+
+    $actual = $section->find('css', '.ecl-footer-core__title');
+    $this->assertEquals('Institution links title', $actual->getText());
+
+    $items = $data['institution_links'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'core', $expected);
+    }
+
+    // Update settings, assert footer changed.
+    $this->updateSiteSettings();
```

Comment:
This is similar to the issue we had outlined in a previous review: the caller needs to pass to the method the values to set, even if it's used once, or else the caller needs to know what that method does.

Change this so that we can set site owner and assert it here below.